lib/shoryuken/queue.rb

```diff
@@ -49,20 +49,24 @@ def fifo?
 
     private
 
-    def set_name(name)
-      self.name = name
-      self.url  = client.get_queue_url(queue_name: name).queue_url
-    rescue Aws::Errors::NoSuchEndpointError, Aws::SQS::Errors::NonExistentQueue => ex
-      raise ex, "The specified queue #{name} does not exist."
-    end
-
-    def set_url(url)
-      self.name = url.split('/').last
-      self.url  = url
+    def parse_queue_url_params(name_or_url)
+      if name_or_url.start_with?('https://sqs.')
+        *_, account_id, queue_name = URI.parse(name_or_url).path.split('/')
+        {
+          queue_name: queue_name,
+          queue_owner_aws_account_id: account_id
+        }
+      else
+        { queue_name: name_or_url }
+      end
     end
 
     def set_name_and_url(name_or_url)
-      name_or_url.start_with?('https://sqs.') ? set_url(name_or_url) : set_name(name_or_url)
+      queue_url_params = parse_queue_url_params(name_or_url)
+      self.name = queue_url_params[:queue_name]
+      self.url  = client.get_queue_url(queue_url_params)
```

Comment:
@maschwenk WDYT about something like [this](https://github.com/phstc/shoryuken/issues/412#issuecomment-316700989) for saving a request, since we need to check `fifo?` anyway, and we don't have to assume URL convention for queues: `*_, account_id, queue_name = URI.parse(name_or_url).path.split('/')`?

Do you mind also adding a test case `queue_spec`?